azurelinuxagent/ga/periodic_operation.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2020 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# Requires Python 2.6+ and Openssl 1.0+
+#
+
+import datetime
+
+from azurelinuxagent.common import logger
+from azurelinuxagent.common.future import ustr
+
+
+class PeriodicOperation(object):
+    '''
+    Instances of PeriodicOperation are tasks that are executed only after the given
```

Comment:
Thanks for adding this class! Will definitely make it easier to setup a periodic operation now! 

NIT: Consider adding a small note here saying that the `run` will never throw so make sure to add your custom try-except if needed

Reply:
Done